File: 80. Remove Duplicates from Sorted Array II/Solution.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length < 2) {
            return nums.length;
        }
        int idx = 0;
        boolean duplicate = false;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] != nums[idx]) {
                duplicate = false;
                idx++;
                nums[idx] = nums[i];
            } else if (!duplicate) {
                idx++;
                nums[idx] = nums[i];
                duplicate = true;
            }
        }
        return idx + 1;
    }
}

File: 80. Remove Duplicates from Sorted Array II/Solution.go
func removeDuplicates(nums []int) int {
	if len(nums) < 2 {
		return len(nums)
	}
    idx := 0
	duplicate := false
	for i := 1; i < len(nums); i++ {
		if nums[i] != nums[idx]    {
			duplicate = false
			idx++
			nums[idx] = nums[i]
		} else if !duplicate {
				idx++
				nums[idx] = nums[i]
                duplicate  = true
	    }
	}
	return idx +1
}